__tests__/commands/install/integration.js

```diff
@@ -61,6 +61,25 @@ test.concurrent('installing with --ignore-optional should not install optional s
   });
 });
 
+test.concurrent(
+  'running install inside a workspace should run the install from the root of the workspace',
+  async () => {
+    await runInstall({}, 'install-workspaces', async (config, reporter): Promise<void> => {
+      const pkgJson = await fs.readJson(`${config.cwd}/workspace/package.json`);
+      pkgJson.dependencies['b'] = 'file:../b';
+      await fs.writeFile(`${config.cwd}/workspace/package.json`, JSON.stringify(pkgJson));
+
+      const workspaceConfig = await Config.create({cwd: `${config.cwd}/workspace`}, reporter);
+
+      const reInstall = new Install({}, workspaceConfig, reporter, await Lockfile.fromDirectory(config.cwd));
+      await reInstall.init();
+
+      expect((await fs.readFile(`${config.cwd}/yarn.lock`)).indexOf(`"b@file:b"`)).not.toEqual(-1);
+      expect((await fs.readFile(`${config.cwd}/yarn.lock`)).indexOf(`"a@file:./a"`)).not.toEqual(-1);
```

Comment:
Awoid reading the same file twice and store the file contents in a variable, please.